backend/lib/fetch/providers.py

```diff
@@ -90,27 +90,31 @@ def fetch_providers(providers, geocoder_name=GEOCODER, engine=connect.create_db_
         len(provider_addresses), len(providers))
     )
 
-    existing_addresses = {
-        result.address: {
-            'id': result.id,
-            'latitude': result.latitude,
-            'longitude': result.longitude
-        } for result in _fetch_addresses_from_db(provider_addresses, session)
-    }
-
-    logger.debug('Found {} addresses in DB out of {}.'.format(
-        len(existing_addresses), len(provider_addresses))
-    )
+    if config.get('address_database'):
+        existing_addresses = {
+            result.address: {
+                'id': result.id,
+                'latitude': result.latitude,
+                'longitude': result.longitude
+            } for result in _fetch_addresses_from_db(provider_addresses, session)
+        }
+        logger.debug('Found {} addresses in DB out of {}.'.format(
+            len(existing_addresses), len(provider_addresses))
+        )
+    else:
+        logger.debug('Address database deactivated.')
```

Comment:
when would this happen? should it be a warning? the scope of this PR seems to have creeped

Reply:
There is config for that. This makes testing much easier, plus it will allow us to have different behavior in the future, for example if we do not want to save everybody's geocoding in our DB. Happy to remove if you think it is not useful

Reply:
can you maybe make the config value a little clearer? just looking at the name i would think that it's maybe a URL to a database of addresses